File: ruby_method/features/steps_definitions/unsuccessful_login.rb
# encoding: utf-8

Given(/^I am in the login page$/) do
  @login_page = LoginPage.new
  @login_page.load
  sleep 3
end

When(/^I set the user (.+) and password (.+)$/) do |user, password|
  @login_page.wait_until_input_username_visible
  @login_page.wait_until_input_password_visible
  @login_page.input_username.set(user)
  @login_page.input_password.set(password)
end

And(/^click in the Sign In button$/) do
  @login_page.btn_sign_in.click
end

Then(/^must to show a wrong user or password warning message$/) do
  @login_page.wait_until_warning_message_visible
  expect(@login_page).to have_warning_message
end

File: python_method/python_method/features/steps/unsuccessful_login.py
from time import sleep
from behave import given, when, then
from selenium import webdriver

from python_method.features.pages.loginPage import LoginPage

@given(u'I am in the login page')
def access_login_page(context):
    link = context.data_env.get('url')
    context.login_page = LoginPage(context.browser, root_uri=link)
    context.login_page.get('/login')
    sleep(5)

@when(u'I set the user and password and click in the Sign In button')
def set_login(context):

    context.login_page.input_username = context.data_env.get('email')
    context.login_page.input_password = context.data_env.get('password')

    context.login_page.btn_sign_in.click()
    sleep(3)

@then(u'must to show a wrong user or password warning message')
def verify_message(context):
    assert(context.login_page.warning_message != None), "The message was: "
    print(context.login_page.warning_message.text)
